spec/services/hmpps_api/community_api_spec.rb

```diff
@@ -0,0 +1,137 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe HmppsApi::CommunityApi do
+  let(:api_host) { Rails.configuration.community_api_host }
+  let(:offender_no) { 'A1234BC' }
+
+  before do
+    stub_auth_token
+  end
+
+  describe '.set_pom' do
+    let(:stub_url) { "#{api_host}/secure/offenders/nomsNumber/#{offender_no}/prisonOffenderManager" }
+
+    before do
+      stub_request(:put, stub_url).to_return(status: 200, body: '{}')
+    end
+
+    it 'sets the POM name for the given offender' do
+      described_class.set_pom(
+        offender_no: offender_no,
+        prison: 'PVI',
+        forename: 'Jane',
+        surname: 'Doe'
+      )
+
+      expect_body = {
+        nomsPrisonInstitutionCode: 'PVI',
+        officer: {
+          forenames: 'Jane',
+          surname: 'Doe'
+        }
+      }
+
+      expect(WebMock).to have_requested(:put, stub_url).
+        with(
+          headers: { 'Content-Type': 'application/json' },
+          body: expect_body.to_json
+        )
+    end
+  end
+
+  describe '.set_handover_dates' do
+    subject {
+      described_class.set_handover_dates(
+        offender_no: offender_no,
+        handover_start_date: handover_start_date,
+        responsibility_handover_date: responsibility_handover_date
+      )
+    }
+
+    let(:stub_base_url) { "#{api_host}/secure/offenders/nomsNumber/#{offender_no}/custody/keyDates" }
+
+    before do
+      stub_request(:put, "#{stub_base_url}/POM1").to_return(status: 200, body: '{}')
+      stub_request(:put, "#{stub_base_url}/POM2").to_return(status: 200, body: '{}')
+      stub_request(:delete, "#{stub_base_url}/POM1").to_return(status: 200)
+      stub_request(:delete, "#{stub_base_url}/POM2").to_return(status: 200)
+
+      # Trigger the request
+      subject
```

Comment:
I think this is a slight misuse of 'subject'. IMHO the 'subject' block should just be here explicitly in the before block

Reply:
Fixed